api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -247,8 +248,8 @@ public boolean maybeSetDollarLimitOverride(DbUser user, double newDollarLimit) {
       user = userDao.save(user.setInitialCreditsLimitOverride(newDollarLimit));
 
       if (userHasRemainingInitialCredits(user)) {
-        // may be redundant: enable anyway
-        setAllToUnexhausted(user);
+        // may be redundant: clear exhaustion anyway
+        updateInitialCreditsExhaustion(user, false);
```

Comment:
make this more general, so it can replace the very similar WorkspaceService.updateInitialCreditsExhaustion()